ios/Classes/SwiftDeviceCalendarPlugin.swift

```diff
@@ -322,11 +322,47 @@ public class SwiftDeviceCalendarPlugin: NSObject, FlutterPlugin, EKEventViewDele
                     let endDate = Date (timeIntervalSince1970: endDateDateMillisecondsSinceEpoch!.doubleValue / 1000.0)
                     let ekCalendar = self.eventStore.calendar(withIdentifier: calendarId)
                     if ekCalendar != nil {
-                        let predicate = self.eventStore.predicateForEvents(
-                            withStart: startDate,
-                            end: endDate,
-                            calendars: [ekCalendar!])
-                        let ekEvents = self.eventStore.events(matching: predicate)
+                        var ekEvents = [EKEvent]()
+                        let fourYearsInSeconds = 4 * 365 * 24 * 60 * 60
+                        var currentStartDate = startDate
+                        // Adding 4 years to the start date
+                        var currentEndDate = startDate.addingTimeInterval(TimeInterval(fourYearsInSeconds))
+                        while currentEndDate <= endDate {
+                            let rangeSize = currentEndDate.timeIntervalSince(currentStartDate)
+                            let roundedRangeSize = Int(rangeSize / Double(fourYearsInSeconds)) * fourYearsInSeconds
```

Comment:
Why do we need to round the `rangeSize`?

As far as I can tell, we only use the `roundedRangeSize` to advance the start / end dates for the next batch. Why can't we just use `fourYearsInSeconds` instead?

And if the `roundedRangeSize` is needed, why do we need to calculate it every iteration?

Reply:
I overcomplicated the calculations a little, it really could have been done much easier 😅
Updated